backend/src/nodes/nodes/stable_diffusion/vae_encode.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import numpy as np
+import torch
+from comfy.latent_image import RGBImage
+
+from ...impl.external_stable_diffusion import nearest_valid_size
+from ...impl.pil_utils import InterpolationMethod, resize
+from ...impl.stable_diffusion.types import LatentImage, VAEModel
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput
+from ...properties.inputs.stable_diffusion_inputs import VAEModelInput
+from ...properties.outputs.stable_diffusion_outputs import LatentImageOutput
+from ...utils.utils import get_h_w_c
+from . import category as StableDiffusionCategory
+
+
+@NodeFactory.register("chainner:stable_diffusion:vae_encode")
+class VAEEncodeNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = ""
+        self.inputs = [
+            ImageInput(channels=3),
+            VAEModelInput(),
+        ]
+        self.outputs = [
+            LatentImageOutput(
+                image_type="""def nearest_valid(n: number) = int & floor(n / 64) * 64;
+                LatentImage {
+                    width: nearest_valid(Input0.width),
+                    height: nearest_valid(Input0.height)
+                }""",
+            ),
+        ]
+
+        self.category = StableDiffusionCategory
+        self.name = "VAE Encode"
+        self.icon = "PyTorch"
+        self.sub = "Input & Output"
+
+    @torch.no_grad()
+    def run(self, image: np.ndarray, vae: VAEModel) -> LatentImage:
+        height, width, _ = get_h_w_c(image)
+
+        width1, height1 = nearest_valid_size(
+            width, height, step=64
+        )  # This cooperates with the "image_type" of the ImageOutput
+
+        if width1 != width or height1 != height:
+            image = resize(image, (width1, height1), InterpolationMethod.AUTO)
+
+        try:
+            vae.cuda()
+            img = RGBImage.from_array(image, device="cuda")
```

Comment:
same thing with the device param here. Might be more complex than just changing the .cuda calls, but can probably just pass in the execution options device

Reply:
So, I went and replaced the `.cuda()`s with `.to(exec_options.full_device)`.  For some reason, that seems to be set to "cpu" for me.  I don't have the CPU Mode setting set in the options.  I'm not really sure how it's getting set.

Reply:
The normal PyTorch nodes seem to be running on GPU.  Odd...

Reply:
My guess is that since this isn't pytorch directly (I'm not even sure what this is) it probably doesn't support that syntax. I'd say just check if the device is CUDA and if so do this, otherwise do CPU

Reply:
The `.to` calls are getting passed transparently into pytorch.  When I stick a breakpoint in the sampler node, `exec_options.full_device` is set to `cpu`.   When I put the breakpoint in the PyTorch load model node, it's set to `cuda`.  I don't really know why that would be, and I haven't had a chance to dig into it.  It looks like it's supposed to be a global.

![image](https://user-images.githubusercontent.com/551173/231089472-380606cf-2f79-4cb1-afbd-46a82df55781.png)
![image](https://user-images.githubusercontent.com/551173/231089615-4b1f4192-8045-45fd-b43c-ec3f365e810b.png)